src/Microsoft.AspNetCore.Mvc.Core/Internal/ActionSelector.cs

```diff
@@ -234,5 +285,217 @@ protected virtual IReadOnlyList<ActionDescriptor> SelectBestActions(IReadOnlyLis
                 return EvaluateActionConstraintsCore(context, actionsWithoutConstraint, order);
             }
         }
+
+        // The action selector cache stores a mapping of route-values -> action descriptors for each 'known' set of
+        // of route-values. We actually build two of these mappings, one for case-sensitive (fast path) and one for
+        // case-insensitive (slow path).
+        //
+        // The difference here is because while MVC is case-insensitive, doing a case-sensitive comparison is much 
+        // faster. We also expect that most of the URLs we process are canonically-cased because they were generated
+        // by Url.Action or another routing api.
+        //
+        // This means that for a set of actions like:
+        //      { controller = "Home", action = "Index" } -> HomeController::Index1()
+        //      { controller = "Home", action = "index" } -> HomeController::Index2()
+        //
+        // There are two entries in the case-sensitive dictionary, each one maps maps to **both** of these action
+        // methods.
```

Comment:
There's no such thing as `case-sensitively` in routing. Routing/action selection is always case-insensitive. We've building a case-sensitive dictionary that will include all case-insensitive matches.